content-src/components/DiscoveryStreamComponents/TopSites/TopSites.jsx

```diff
@@ -4,14 +4,138 @@
 
 import { connect } from "react-redux";
 import { TopSites as OldTopSites } from "content-src/components/TopSites/TopSites";
+import { TOP_SITES_MAX_SITES_PER_ROW } from "common/Reducers.jsm";
 import React from "react";
 
 export class _TopSites extends React.PureComponent {
+  // Find a SPOC that doesn't already exist in User's TopSites
+  getFirstAvailableSpoc(topSites, data) {
+    const { spocs } = data;
+    if (!spocs || spocs.length === 0) {
+      return null;
+    }
+
+    const userTopSites = new Set(
+      topSites.map(topSite => topSite && topSite.url)
+    );
+
+    // We "clean urls" with http in TopSiteForm.jsx
+    // Spoc domains are in the format 'sponsorname.com'
+    return spocs.find(
+      spoc =>
+        !userTopSites.has(spoc.url) &&
+        !userTopSites.has(`http://${spoc.domain}`) &&
+        !userTopSites.has(`http://www.${spoc.domain}`)
+    );
+  }
+
+  // Find the first empty or unpinned index we can place the SPOC in.
+  // Return -1 if no available index and we should push it at the end.
+  getFirstAvailableIndex(topSites, promoAlignment) {
+    if (promoAlignment === "left") {
```

Comment:
For left alignment, organic sites menu option edit is broken

Reply:
😅, nice bug catch. 

It looks like for the Edit Option we pass in the site data all the way from the parent TopSites in AS, which will use the original TopSites.

https://github.com/mozilla/activity-stream/blob/d9ef50f2db4d0be7b7d31a757687586b78e061c1/content-src/components/TopSites/TopSites.jsx#L180-L185